src/main/java/com/exasol/adapter/dynamodb/queryplan/OrPredicate.java

```diff
@@ -0,0 +1,35 @@
+package com.exasol.adapter.dynamodb.queryplan;
+
+import java.util.List;
+
+/**
+ * This class represents a {@code OR} predicate.
+ */
+@java.lang.SuppressWarnings("squid:S119") // DocumentVisitorType does not fit naming conventions.
+public class OrPredicate<DocumentVisitorType> implements DocumentQueryPredicate<DocumentVisitorType> {
```

Comment:
This class looks quite similar to AndPrediacte. Maybe you mark instances with type AND/OR field and shrink that to one class?